### ISSUE TITLE
Expat Build error

### ISSUE THREAD

**Issue by DubStepMad** [Lihis/ets2-job-logger#9]
`Expat/2.2.6@pix4d/stable:
Expat/2.2.6@pix4d/stable: ERROR: Package 'eb4102e772b530edfbc4b0e1be0c1f346423e000' build failed
Expat/2.2.6@pix4d/stable: WARN: Build folder C:\Users\{USERNAME}\.conan\data\Expat\2.2.6\pix4d\stable\build\eb4102e772b530edfbc4b0e1be0c1f346423e000
ERROR: Expat/2.2.6@pix4d/stable: Error in build() method, line 45
        cmake.configure(source_dir="../libexpat/expat", build_dir="build", defs=cmake_args)
        ConanException: Error 1 while executing cd build && cmake -G "Visual Studio 16 2019" -A "x64" -DCONAN_LINK_RUNTIME="/MDd" -DCONAN_IN_LOCAL_CACHE="ON" -DCONAN_COMPILER="Visual Studio" -DCONAN_COMPILER_VERSION="16" -DCONAN_CXX_FLAGS="/MP16" -DCONAN_C_FLAGS="/MP16" -DBUILD_SHARED_LIBS="OFF" -DCMAKE_INSTALL_PREFIX="C:\Users\{USERNAME}\.conan\data\Expat\2.2.6\pix4d\stable\package\eb4102e772b530edfbc4b0e1be0c1f346423e000" -DCMAKE_INSTALL_BINDIR="bin" -DCMAKE_INSTALL_SBINDIR="bin" -DCMAKE_INSTALL_LIBEXECDIR="bin" -DCMAKE_INSTALL_LIBDIR="lib" -DCMAKE_INSTALL_INCLUDEDIR="include" -DCMAKE_INSTALL_OLDINCLUDEDIR="include" -DCMAKE_INSTALL_DATAROOTDIR="share" -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY="ON" -DCONAN_EXPORTED="1" -Wno-dev -DBUILD_doc="OFF" -DBUILD_examples="OFF" -DBUILD_shared="False" -DBUILD_tests="OFF" -DBUILD_tools="OFF" -DCMAKE_POSITION_INDEPENDENT_CODE="ON" -DCMAKE_DEBUG_POSTFIX="" -DMSVC_USE_STATIC_CRT="False" ../libexpat/expat`

**Comment by Lihis** [comment]
I would say that this is not our issue but rather Conan/Expat issue related to Visual Studio 2019 as with Visual Studio 2017 there is no problems. Will you be able to resolve this by yourself? If it is issue on Conan or Expat feel free to link the related issue. Or open pull request if we need to change something to make it work with VS2019.

**Comment by DubStepMad** [comment]
> I would say that this is not our issue but rather Conan/Expat issue related to Visual Studio 2019 as with Visual Studio 2017 there is no problems. Will you be able to resolve this by yourself? If it is issue on Conan or Expat feel free to link the related issue. Or open pull request if we need to change something to make it work with VS2019.

Thank you for the quick response and I'll see if there are any fixes available otherwise I will install VS2015.

Thank you for your help.

**Comment by DubStepMad** [comment]
@Lihis Not sure which package you are using but found out that the main Expat is now deprecated: https://github.com/bincrafters/conan-expat

Updated version is: https://github.com/libexpat/libexpat

I hope this helps to resolve the issues.

**Comment by Lihis** [comment]
`wxWidgets` seems to be the one pulling in the `Expat` as a dependency. Does it work with VS2019 if you make Conan to pull in the latest stable release of Expat a.k.a `2.2.9`?

Answering the question about VS2015 in the different issue; like I said above, building should work in Visual Studio **2017**.

**Comment by DubStepMad** [comment]
@Lihis installed VS2017 and the CMake extentsion but now getting a different error:

`Expat/2.2.6@pix4d/stable: WARN: Can't find a 'Expat/2.2.6@pix4d/stable' package for the specified settings, options and dependencies:
- Settings: arch=x86_64, build_type=Debug, compiler=Visual Studio, compiler.runtime=MDd, compiler.version=16, os=Windows
- Options: disable_getrandom=True, shared=False, static_crt=False
- Dependencies:
- Package ID: eb4102e772b530edfbc4b0e1be0c1f346423e000`

**Comment by donatocl** [comment]
If help, I compiled the gui and plugin like 10 minutes ago without problems, following the instructions on readme for windows. Using Visual Studio 2017 x64 + Visual Studio Tools. As said in the instructions,  I used conan to install dependencies.

**Comment by Lihis** [comment]
That's good. Unfortunately I can't help much with Windows stuff so I'll edit the title to indicate that this is VS2019 specific issue and leave this open for now.

**Comment by DubStepMad** [comment]
> If help, I compiled the gui and plugin like 10 minutes ago without problems, following the instructions on readme for windows. Using Visual Studio 2017 x64 + Visual Studio Tools. As said in the instructions, I used conan to install dependencies.

I've followed the instructions and used VS2017 but still have issues.

**Comment by Lihis** [comment]
Comparing the output from AppVeyor build and what you had pasted above and I see the `compiler.version` is different. In AppVeyor it is `15` while you have `16`. Is it so that version `16` is shipped with VS2019 while version `15` is shipped with VS2017?

Also the error message in the first message does not tell much, except execution failed, but who knows why it failed, there wasn't any details?

**Comment by DubStepMad** [comment]
@Lihis 
I didn't want to open another issue as this is still related to Windows 10 but using VS 2017

After doing the following step `Edit CMake options via CMake -> Change CMake settings -> CMakelists.txt`

These are the options that is prompted, which one am I suppose to select?
![rXz32](https://user-images.githubusercontent.com/13799856/74347035-630f1a00-4da8-11ea-886b-eb3ecf16d390.png)

**Comment by DubStepMad** [comment]
Even with the CMake GUI version it has an issue finding a file and with a command.
![eYTXW](https://user-images.githubusercontent.com/13799856/74351011-702f0780-4dae-11ea-9c38-c438b79efae5.png)

**Comment by donatocl** [comment]
@DubStepMad hey, you need to select x64-Release and then, follow the intructions that are inside the readme for windows build.

**Comment by Lihis** [comment]
> hey, you need to select x64-Release and then, follow the intructions that are inside the readme for windows build.

If one follows the instructions in readme shouldn't they choose `x64-Debug` as it guides for debug build?

**Comment by donatocl** [comment]
> 
> 
> > hey, you need to select x64-Release and then, follow the intructions that are inside the readme for windows build.
> 
> If one follows the instructions in readme shouldn't they choose `x64-Debug` as it guides for debug build?

Yes, x64-Debug if they want to build a debug build or release for a release build. Also they can choose x64-Debug and as is explained in the readme, after the debug release works, they can replace all the debug for release in the cmake-settings and in the conan install command line.
